torch/csrc/jit/passes/quantization.cpp

```diff
@@ -463,17 +426,26 @@ void InsertQuantDequantNodes(
   }
 
   // Insert the quant-dequant pair for values output from quantizable nodes
-  for (auto& ele : quantOutputs) {
-    if (qparam_value_dict.count(ele) != 0) {
-      addQuantDeQuantNodes(ele, qparam_value_dict[ele]);
+  for (auto& v_to_quant : quantOutputs) {
+    if (qparam_value_dict.count(v_to_quant) != 0) {
+      Node* dq = addQuantDeQuantNodesFor(
+          v_to_quant, v_to_quant->node(), qparam_value_dict[v_to_quant]);
+      TORCH_INTERNAL_ASSERT(dq != nullptr);
+      v_to_quant->replaceAllUsesWith(dq->output());
+      // Above step replaces v->quant with vdq->quant. We need to restore link.
+      // Below chain traverse up from dq to q node.
+      dq->inputs()[0]->node()->inputs()[0]->node()->replaceInputWith(
```

Comment:
Maybe factor `dq->inputs()[0]->node()->inputs()[0]->node()` out to a separate function and add asserts there?

Reply:
Done